docs/DeveloperGuide.md

```diff
@@ -1,14 +1,155 @@
 # Developer Guide
 
-## Design & implementation
+* Table of Contents
+{:toc} 
 
+## Design 
+
+### UI component
+![Ui component](diagrams/Ui component.png)
+
+**API**: [`Ui.java`](https://github.com/AY2021S1-CS2113-T14-4/tp/blob/master/src/main/java/seedu/dietbook/Ui.java)
+
+The `UI` component,
+* Takes in user command and passes to the `Logic` components for command execution.
+* Updates the user about any changes in the data after executing the command or errors encountered when executing the commands.
+
+The UI has a dependency with two enumeration class, [`ActivityLevel`](https://github.com/AY2021S1-CS2113-T14-4/tp/blob/master/src/main/java/seedu/dietbook/person/ActivityLevel.java) and [`Gender`](https://github.com/AY2021S1-CS2113-T14-4/tp/blob/master/src/main/java/seedu/dietbook/person/Gender.java) as descriptions of each [`ActivityLevel`](https://github.com/AY2021S1-CS2113-T14-4/tp/blob/master/src/main/java/seedu/dietbook/person/ActivityLevel.java) and [`Gender`](https://github.com/AY2021S1-CS2113-T14-4/tp/blob/master/src/main/java/seedu/dietbook/person/Gender.java) is required. Increased coupling was sacrificed to reduce code duplicates and increase ease of code extension/editing.
+
+## Implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
```

Comment:
Perhaps you should delete this line for the final version?

Reply:
Noted! Thank you!